redis/cache.go

```diff
@@ -0,0 +1,84 @@
+package redis
+
+import (
+	"fmt"
+	"github.com/cenkalti/backoff"
+	"github.com/gomodule/redigo/redis"
+	"github.com/mna/redisc"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+var (
+	// ConnBackoffTimeout is the duration before the backoff will timeout
+	ConnBackoffTimeout = time.Duration(30) * time.Second
+)
+
+type Cache interface {
+	Delete(key string) (string, error)
+	Get(key string) (string, error)
+	Exists(key string) (bool, error)
+	Set(key, val string) error
+	SetExpiry(key, val string, expireTime int) error
+	SetExpiryTime(key, val string, expireTime time.Time) error
+	Subscribe(subscription string, handleResponse func(interface{}))
+}
+
+type cache struct {
+	Pool *redisc.Cluster
```

Comment:
make it local, there is no need for it to be accessible outside

Reply:
thanks